### PR TITLE
Publish Docker images to public ECR as part of the release-staging workflow

### PR DESCRIPTION
**Pull request by emedvedev**
Completely untested, so WIP. From what I understand, more or less what we need.

FYI I've requested `chainlink-staging` as a custom alias for our public ECR registry on staging, pending approval. Same for `chainlink` on prod.

### CODE REVIEW

**Comment by emedvedev** [comment]
@HenryNguyen5 this is now fully tested:
1. You can see the release run here: https://github.com/smartcontractkit/external-adapters-js/actions/runs/428109879
2. You can check the images in the Public ECR: https://us-west-2.console.aws.amazon.com/ecr/repositories?region=us-west-2
3. This PR was applied to staging to grant Public ECR permissions for CI/CD: https://github.com/smartcontractkit/infra-staging/pull/342
Ready to be merged whenever you want.

File: .github/workflows/release-staging.yml [hunk 1]
```diff
@@ -7,6 +7,7 @@ name: Release Staging
 
 env:
   cache-name: staging-cd
+  publicecr-name: x5x8k3n7
```
Comment (krebernisak) [inline .github/workflows/release-staging.yml:10]: Should we be able to put the alias here when it's approved?

**Comment by krebernisak** [comment]
Thank you @emedvedev, this should be very useful 👍 

I think we should also have a repo for our [release workflow](https://github.com/smartcontractkit/external-adapters-js/blob/716bf327c7ff2ac25edde0e35c39485d4dce2974/.github/workflows/release.yml). @HenryNguyen5 made two S3 buckets for zips, staging & release. We should probably do the same for Docker images.

File: .github/workflows/release-staging.yml (hunk 1 above)
Comment (emedvedev) [reply .github/workflows/release-staging.yml:10]: Yes! No idea when that could happen though, Amazon has only rolled out Public ECR two weeks ago so there isn't much information. Hopefully, it'll be done by the time we all come back from the holidays.

**Comment by emedvedev** [comment]
@krebernisak oh, absolutely, just wanted to test on staging first. :) I've already created the public registry for production (and requested an alias), just need to grant CI/CD permissions to publish there and then update the workflow. I'm not sure I'll have time for that today, so most likely we'll pick that up right after the holidays.

**Comment by emedvedev** [comment]
@HenryNguyen5 feel free to review and merge this, and I'll open another PR for production next week if this one's fine :)

**Comment by emedvedev** [comment]
`chainlink-staging` and `chainlink` aliases for our registries were both approved (changed the staging name in the PR, too), so now we can have nice URLs like `https://public.ecr.aws/chainlink-staging/adapters/amberdata-adapter` ❤️

**Comment by emedvedev** [comment]
@HenryNguyen5 added the release workflow steps as well, so should be ready to merge now.

The infra PR to grant the workflow prod access to Public ECR should be merged first: https://github.com/smartcontractkit/infra/pull/970

**Review by krebernisak — DISMISSED** [review]
Awesome! Thanks for your work on this @emedvedev & @HenryNguyen5 🙌

**Comment by emedvedev** [comment]
@HenryNguyen5 merged the latest master, ready for review

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -8,6 +8,7 @@ name: Release
 
 env:
   cache-name: release-ci
+  publicecr-name: chainlin
```
Comment (HenryNguyen5) [inline .github/workflows/release.yml:11]:
```suggestion
  publicecr-name: chainlink
```

**Review by HenryNguyen5 — CHANGES_REQUESTED** [review]

**Review by HenryNguyen5 — APPROVED** [review]
Looks great, thanks @emedvedev !

**Pull request merged**